packages/atoms/connect/ConnectButton.tsx

```diff
@@ -0,0 +1,55 @@
+import { useState } from "react";
+
+import { classNames } from "@calcom/lib";
+
+import { Button } from "../src/components/ui/button";
+import type { AtomsGlobalConfigProps } from "../types";
+import type { ConnectButtonProps } from "../types";
+
+// we accept two props: buttonText & onButtonClick
+// buttonText is for displaying buttonText
+// onButtonClick is a a method to handle onClick for button
+// it accepts another object which contains a callback in case of any error
+
+export function ConnectButton({
+  buttonText,
+  onButtonClick,
+  stylesClassname,
+  icon,
+}: ConnectButtonProps & AtomsGlobalConfigProps) {
+  const [isProcessing, setIsProcessing] = useState<boolean>(false);
+  const [errMsg, setErrMsg] = useState<string>("");
+
+  function handleSubmit(e: React.MouseEvent<HTMLButtonElement, MouseEvent>) {
+    e.preventDefault();
+
+    setIsProcessing(true);
+    onButtonClick({
+      cb: (err) => {
+        setIsProcessing(false);
+        if (err?.error.message) {
+          setErrMsg(err.error.message);
+        }
+      },
+    });
+  }
+
+  return (
+    <div>
+      <Button
+        className={classNames(
```

Comment:
Hey there, I think for cal atoms we should try to follow a similar approach as shown in shadcn/ui

meaning using the `cn` util and class variance authority to handle different variants of the component